contrib_bots/lib/xkcd.py

```diff
@@ -0,0 +1,120 @@
+import requests
+import logging
+from random import randint
+
+class xkcdHandler(object):
+    '''
+    This plugin provides several command that can be used for fetch comics from
+    xkcd. The bot looks for messages starting with "@xkcd" and responds with a
+    message with the comic based on provided commands.
+    '''
+
+    XKCD_TEMPLATE = 'http://xkcd.com/%s/info.0.json'
+    LATEST_XKCD = 'http://xkcd.com/info.0.json'
+
+    def usage(self):
+        return '''
+            This plugin allows users to fetch comics provided by xkcd.
+            Users should preface the command with "@xkcd".
+
+            There are several commands to use this bot:
+            - @xkcd help -> To show all commands the bot supports.
+            - @xkcd latest -> To fetch latest comic from xkcd.
+            - @xkcd random -> To fetch random comic from xkcd.
+            - @xkcd <comic_id> -> To fetch comic based on `<comic_id>`, e.g
+              `@xkcd 1234`.
+            '''
+
+    def fetch_xkcd_query(self, mode, comic_id=None):
+        try:
+            if mode == 0:  # Fetch latest comic.
```

Comment:
If you wanted this to be more readable I would suggest using an enum:

https://docs.python.org/3/library/enum.html

```
class Command(Enum):
    LATEST = 0
    ...
```

Like `Command.LATEST` could be the thing to look for here. By the ways, enums are available in Python since version 3.4. If we want to stay Python 2 compatible (don't know if we do for contrib bots) you could have the equivalent functionality with the following class:

```
class Command():
     LATEST = 0
     ....
```

This doesn't get you some of the fancy stuff that comes with `enum` though.